[android] better user profile (initial layout)

RedAuburn opened the pull request:
Some work towards a better user profile page. 

Sadly i'm gonna be really busy for the next couple of months, so loading the profile image will have to wait (although it's somewhat working at [redauburn/organicmaps/android-user-profile](https://github.com/RedAuburn/organicmaps/tree/android-user-profile), albeit in the wrong place...)

i thought i may as well submit this stripped down version though, which has a new user info layout, displays the username, and formats the changeset count according to locale.

before/after:

![cvbfdfdfg](https://github.com/organicmaps/organicmaps/assets/26939824/201099ea-f046-486d-b75b-53b5bb9232a1)


generic profile image from https://svgsilh.com/image/378368.html (CC0)

biodranik commented on android/app/build.gradle:
```diff
@@ -393,6 +393,8 @@ dependencies {
   implementation 'com.github.devnullorthrow:MPAndroidChart:3.2.0-alpha'
   implementation 'net.jcip:jcip-annotations:1.0'
+  // Used for circular profile image in profile
+  implementation 'de.hdodenhof:circleimageview:3.1.0'
```
Would a square image be enough? It would be great to avoid non-critical dependencies.

biodranik commented on android/app/src/main/java/app/organicmaps/editor/ProfileFragment.java:
```diff
@@ -61,14 +68,25 @@ private void refreshViews()
+          mProfileName.setText(profileUsername);
+
+          // use generic img if user has no pfp or couldn't load
+          if(profilePicture != null)
```
```suggestion
          if (profilePicture != null)
```

biodranik commented on android/app/src/main/res/drawable/profile_generic.xml:
```diff
@@ -0,0 +1,7 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android" android:height="105.34979dp" android:viewportHeight="1280" android:viewportWidth="1215" android:width="100dp">
```
How a user Material icon looks compared to this one? https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Aperson%3AFILL%401%3Bwght%40400%3BGRAD%400%3Bopsz%4024

biodranik reviewed (commented):
Thanks! That is a good start.

Jean-BaptisteC commented on android/app/src/main/res/layout/fragment_osm_profile.xml:
```diff
@@ -24,103 +24,108 @@
+          android:id="@+id/user_profile_image"
+          android:layout_width="100dp"
+          android:layout_height="100dp"
+          android:src="@drawable/profile_generic"
```
```suggestion
          app:srcCompat="@drawable/profile_generic"
```

Jean-BaptisteC reviewed (approved):
✅ Pixel 6 - Android 14
Just need improvement when size police is more big
![Screenshot_20240517-000714](https://github.com/organicmaps/organicmaps/assets/87148630/7bc03d67-1060-4984-b03c-1f7adcc59d51)

RedAuburn commented:
fixed scaling, using new image, removed circle image view

![image1](https://github.com/organicmaps/organicmaps/assets/26939824/18b1ab7e-dc44-4640-94ed-0e9285dc346c)

biodranik commented:
1. Why is the image not scaled when text is large? The real picture also won't scale, right?
2. Should user be able to set/change his profile photo after a tap on the stub image?

RedAuburn commented:
> 1. Why is the image not scaled when text is large? The real picture also won't scale, right?

it won't, i didn't think it needed to be bigger but i'll make it scale to fit.

> 2. Should user be able to set/change his profile photo after a tap on the stub image?

Definitely, however it makes more sense to implement this when displaying a user profile picture is actually done, otherwise it'll confuse users

RedAuburn commented:
Converted to ConstraintLayout, profile icon now scales properly and it generally works much better now

biodranik commented:
Thanks! Can the title (OpenStreetMap Profile) be somehow autoscrolled when its too large? How Android usually solve this issue in other places?

RedAuburn commented:
Reworked again to make text always fit (looks mostly the same)

> Thanks! Can the title (OpenStreetMap Profile) be somehow autoscrolled when its too large? How Android usually solve this issue in other places?

This is an issue with headers across the app, so I'll open a separate issue for it. There's a marquee scroll effect which we could use, but maybe a better fix would be to use a shorter string when there are larger font sizes (OpenStreetMap Profile -> OSM Profile)

biodranik commented on android/app/src/main/java/app/organicmaps/editor/ProfileFragment.java:
```diff
@@ -61,14 +68,25 @@ private void refreshViews()
+          mEditsSent.setText(NumberFormat.getInstance().format(profileEditCount));
+          mProfileName.setText(profileUsername);
+
+          // use generic img if user has no pfp or couldn't load
```
```suggestion
          // Use generic image if user has no profile picture or it failed to load.
```

Is image loading repeated somehow in the case of some recoverable errors?

RedAuburn replied:
it should be 👍️

biodranik reviewed (approved):
LGTM with a few nits.

RedAuburn commented:
fixed nits :)

biodranik reviewed (approved):
Thanks!
